### PR TITLE
Feat/update csv download

### PR DESCRIPTION
**Pull request by lissavxo**
Related to #848 

Description
---
Added a selector in CSV export to split the transaction by currency

Test plan
---
- Test export csv all currencies and single currency

### CODE REVIEW

**Review by Klakurka — CHANGES_REQUESTED** [review]
- USD/CAD is going to be an account setting. What I meant here was to have BCH and XEC split out.
- The dropdown label appears in the dropdown list but isn't clickable: 
image - you can just add a `hidden` attribute.
- Cursor should be the pointer when hovering over the dropdown buttons.

**Review by Klakurka — CHANGES_REQUESTED** [review]
- Only show a currency in the dropdown if the button actually has an address for that currency.
- Only show the dropdown at all if the button has addresses for 2 or more currencies.

File: constants/index.ts [hunk 1]
```diff
@@ -166,6 +166,8 @@ export const NETWORK_TICKERS: KeyValueT<string> = {
   bitcoincash: 'BCH'
 }
 
+export type NetworkTickersType = 'XEC' | 'BCH'
```
Comment (Klakurka) [inline constants/index.ts:169]: I'm pretty wary of stuff like this but we can leave it if Estevao thinks it's fine.

I made this phase 3 task though: https://github.com/PayButton/paybutton-server/issues/859

**Review by chedieck — CHANGES_REQUESTED** [review]
Looks good, only some small changes in the code.

As for the pointer cursor in the dropdown, it would require to refactor the dropdown as something other than an actual `<select>` dropdown (such as a list of `<li>` elements) because using CSS to set the cursor as pointer in an actual dropdown doesn't work: the dropdown is rendered by the operating system, not the browser.

We could just ignore it but TBH I think would be way nicer UX if the cursor was really a pointer, since the experience of selecting something that is actually a button feels weird.

**Review by Klakurka — DISMISSED** [review]

File: pages/button/[id].tsx [hunk 2]
```diff
@@ -94,36 +96,52 @@ export default function Button (props: PaybuttonProps): React.ReactElement {
...
   useEffect(() => {
     void getDataAndSetUpSocket()
   }, [])
 
-  const downloadCSV = async (paybutton: { id: string }): Promise<void> => {
+  const downloadCSV = async (paybutton: { id: string, name: string }, currency: string): Promise<void> => {
     try {
-      const response = await fetch(`/api/paybutton/download/transactions/${paybutton.id}`)
+      let url = `/api/paybutton/download/transactions/${paybutton.id}`
+      const isCurrencyEmpty = (value: string): boolean => (value === '')
```
[the remark this answers is not in the document]
Comment (lissavxo) [reply pages/button/[id].tsx]: because I am using it in two places and I changed a bit to add undefined check too

**Review by Klakurka — DISMISSED** [review]
Looks good to me.

We just need a way to get CAD working even if it's by manually setting it in the db for the account that generates the CSV.

File: pages/api/paybutton/download/transactions/[paybuttonId].ts [hunk 3]
```diff
@@ -81,24 +88,32 @@ const formatNumberHeaders = (headers: string[], currency: string): string[] => {
...
-  }).map(tx => {
+  currency: SupportedQuotesType,
+  networkTicker?: NetworkTickersType): Promise<void> => {
+  let networkIdArray = Object.values(NETWORK_IDS)
+  if (networkTicker !== undefined) {
+    const slug = Object.keys(NETWORK_TICKERS).find(key => NETWORK_TICKERS[key] === networkTicker)
+    const networkId = await getNetworkIdFromSlug(slug ?? NETWORK_TICKERS.ecash)
+    networkIdArray = [networkId]
+  }
+  const transactionsGrouped = await fetchTransactionsByPaybuttonIdGroupedByNetwork(paybutton.id, networkIdArray)
```
Comment (chedieck) [inline pages/api/paybutton/download/transactions/[paybuttonId].ts:99]: Why do you need them grouped if you just ungroup it afterwards? This variable is not used anywhere further.

File: services/transactionService.ts [hunk 4]
```diff
@@ -489,6 +500,24 @@ export async function fetchTransactionsByPaybuttonId (paybuttonId: string): Prom
   return transactions
 }
 
+export async function fetchTransactionsByPaybuttonIdGroupedByNetwork (
```
Comment (chedieck) [inline services/transactionService.ts:503]: This is created to be used in a place where the grouping is not relevant. I don't get why it exists, but in any case would be better if it took the txs list, not the button and networkids as a parameter.

**Review by chedieck — CHANGES_REQUESTED** [review]

File: pages/api/paybutton/download/transactions/[paybuttonId].ts (hunk 3 above)
Comment (lissavxo) [reply pages/api/paybutton/download/transactions/[paybuttonId].ts:99]: this is to separate/group by network

File: services/transactionService.ts (hunk 4 above)
Comment (lissavxo) [reply services/transactionService.ts:503]: grouping is relevant, we order by date and group by network.. this way we separate the txs list.

**Review by Klakurka — APPROVED** [review]

**Pull request merged**